feat(Contributing): link widgets with local version of components/sdk-adapter (#103)
webex/widgets: CONTRIBUTING.md

| Commit with type `feat`            | Minor release |
| Commit with type `fix`             | Patch release |
| Commit with type `perf`            | Patch release |

> ## Link widgets with local version of components/sdk-component-adapter

Reviewer: @lalli-flores this requires your review to make sure the explanation is clear and in proper English.

> 

Reviewer: I would add a motivation as a first paragraph, something like:

> When contributing to one of the dependent repositories like `components` or the `sdk-component-adapter`, it may be helpful to be able to test the changes done to the dependent repo in a "real-life" scenario like the meeting widget. In order to do that, you would have to check out both widgets and the dependent repo, and list the dependent package as a local dependency (see https://docs.npmjs.com/cli/v7/configuring-npm/package-json#local-paths).

Reviewer: Below you can find an example for linking a local `components` package to the widgets.

Author: done

First, in your local components/sdk-component-adapter folder, you need to run `npm run build` to generate the "dist"
folder which will be used by widgets.

Then, in your local widgets folder, do the following steps (ex. for components):

- in package.json change

>   `"@webex/components": "^1.79.0",` with

Reviewer: with -> to

Author: done

  `"@webex/file:../components"` (your local components path)
- delete widgets/node_modules/@webex/components folder
- run `npm install`
- run `npm run link ../components/node_modules/react` (this is required ONLY for components)
- run `npm run start`

> Keep in mind that for every modification made in components/sdk-component-adapter, you need to run `npm run build` in

Reviewer: you need to re-run `npm run build` in the dependent repo.

Author: done

| Commit with type `feat`            | Minor release |
| Commit with type `fix`             | Patch release |
| Commit with type `perf`            | Patch release |

> ## Link widgets with local version of components/sdk-component-adapter

Reviewer 2: ```suggestion
## Link Widget repository with local version of components and/or sdk-component-adapter
```

> When contributing to one of the dependent repositories like components or the

Reviewer 2: ```suggestion
When contributing to one of the dependent repositories like [components](https://github.com/webex/components#webex-components) or the
```

> sdk-component-adapter, it may be helpful to be able to test the changes done

Reviewer 2: ```suggestion
[sdk-component-adapter](https://github.com/webex/sdk-component-adapter), it may be helpful to be able to test changes
```

> to the dependent repo in a "real-life" scenario like the meeting widget. In

Reviewer 2: ```suggestion
in the dependent repository on a "real-life" scenarios like in the Meeting widget.
```

> order to do that, you would have to check out both widgets and the dependent

Reviewer 2: ```suggestion
In order to test such changes:
1. Clone widgets and the dependent repositories
2. List the dependent package as a [local dependency](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#local-paths)
```

repo, and list the dependent package as a local dependency
(see https://docs.npmjs.com/cli/v7/configuring-npm/package-json#local-paths).

> Below you can find an example for linking a local `components` package to the widgets.

Reviewer 2: ```suggestion
Below you can find sample steps for linking a local `components` package to the `widgets` repository:
```

> First, in your local `components` folder, you need to run `npm run build` to generate the "dist"

Reviewer 2: ```suggestion
1. Clone `components` repository
    ``bash
     git clone git@github.com:webex/components.git
    ````
2. Build `components` code. This will generate a `dist` folder
    ```bash
    cd components
    npm install
    npm run build
    ```
```

folder which will be used by widgets.

> Then, in your local widgets folder, do the following steps (ex. for components):

Reviewer 2: ```suggestion
3. Change dependency location in `package.json`
    ```
    "@webex/components": "file:../components", // Or corresponding path to local clone
     ```
4. Install dependencies of `widgets` repository
    ```
    npx npm-install-peers
    ```
5. Start up the widget sample
    ```
    npm run start
    ```
```

- in package.json change
  `"@webex/components": "^1.79.0"` to
  `"@webex/components": "file:../components"` (your local components path)
- delete widgets/node_modules/@webex/components folder
- run `npm install`

> - run `npm link ../components/node_modules/react` (this is required ONLY for components)

Reviewer 2: why is this necesssary?

Author: This is necessary because both components and widgets uses React so you'll get an error for having 1 or more copies of React. This command tells to widgets to use the React from components.

[sdk-component-adapter](https://github.com/webex/sdk-component-adapter),
it may be helpful to be able to test changes
in the dependent repository on "real-life" scenarios like in the Meeting widget.

In order to test such changes:
1. Clone widgets and the dependent repositories
2. List the dependent package as a
[local dependency](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#local-paths)

Below you can find sample steps for linking a local `components` package to the `widgets` repository:

1. Clone `components` repository.
    ```bash
    git clone git@github.com:webex/components.git
    ```
2. Build `components` code. This will generate a `dist` folder
    ```bash
    cd components
    npm install
    npm run build
    ```

3. Change dependency location in `package.json`
    ```
    "@webex/components": "file:../components", // Or corresponding path to local clone
    ```
5. Install dependencies of `widgets` repository and the react dependency (this is required because`components` repository uses react)
    ```
    npx npm-install-peers
    npm link ../components/node_modules/react
    ```
6. Start up the widget sample
    ```
    npm run start
    ```

> - run `` (this is required ONLY for components)

Reviewer 2: I think something went wrong here?

Author: Yes, it's some leftover text, I'll remove it.
